azure-kusto-data/azure/kusto/data/_token_providers.py

```diff
@@ -281,81 +277,44 @@ def _init_impl(self):
-        :return: refresh token
-        """
-        import os
+            self._az_token = self._az_auth_context.get_token(self._kusto_uri)
+            return {TokenConstants.AZ_TOKEN_TYPE: TokenConstants.BEARER_TYPE, TokenConstants.AZ_ACCESS_TOKEN: self._az_token.token}
+        except Exception as e:
+            raise KustoClientError(
+                "Failed to obtain Az Cli token for '{0}'.\nPlease be sure AzCli version 2.3.0 and above is intalled.\n{1}".format(self._kusto_uri, e)
+            )
 
+    async def _get_token_impl_async(self) -> Optional[dict]:
         try:
-            # this makes it cleaner, but in case azure cli is not present on virtual env,
-            # but cli exists on computer, we can try and manually get the token from the cache
-            from azure.cli.core._profile import Profile
-            from azure.cli.core._session import ACCOUNT
-            from azure.cli.core._environment import get_config_dir
-
-            azure_folder = get_config_dir()
-            ACCOUNT.load(os.path.join(azure_folder, "azureProfile.json"))
-            profile = Profile(storage=ACCOUNT)
-            token_data = profile.get_raw_token()[0][2]
+            if self._az_auth_context_async is None:
+                self._az_auth_context_async = AsyncAzureCliCredential()
 
-            return token_data
+            self._az_token = await self._az_auth_context_async.get_token(self._kusto_uri, self._az_kwargs)
+            return {TokenConstants.AZ_TOKEN_TYPE: TokenConstants.BEARER_TYPE, TokenConstants.AZ_ACCESS_TOKEN: self._az_token.token}
+        except Exception as e:
+            raise KustoClientError(
+                "Failed to obtain Az Cli token for '{0}'.\nPlease be sure AzCli version 2.3.0 and above is installed.\n{1}".format(self._kusto_uri, e)
+            )
 
-        except ModuleNotFoundError:
-            try:
-                import os
-                import json
+    def _get_token_from_cache_impl(self) -> dict:
+        if self._az_token is not None:
+            # A token is considered valid if it is due to expire in no less than 10 minutes
+            cur_time = time.time()
+            if (self._az_token.expires_on - 600) > cur_time:
+                return {TokenConstants.MSAL_TOKEN_TYPE: TokenConstants.BEARER_TYPE, TokenConstants.MSAL_ACCESS_TOKEN: self._az_token.token}
 
-                folder = os.getenv("AZURE_CONFIG_DIR", None) or os.path.expanduser(os.path.join("~", ".azure"))
-                token_path = os.path.join(folder, "accessTokens.json")
-                with open(token_path) as f:
-                    data = json.load(f)
+        return None
 
-                # TODO: not sure I should take the first
-                return data[0]
-            except Exception as e:
-                raise KustoClientError("Azure cli token was not found. Please run 'az login' to setup account.", e)
+    async def _get_token_from_cache_impl_async(self) -> Optional[dict]:

```

Comment:
This method is not needed - the default behavior of _get_token_from_cache_impl_async is to call the sync version